core/src/commitment.rs

```diff
@@ -123,6 +123,22 @@ impl BlockCommitmentCache {
         self.root
     }
 
+    fn highest_slot_with_confirmation(&self, confirmation_count: usize) -> Slot {
```

Comment:
I think this should be computed ahead of time like how we handle `largest_confirmed_root` because it's kinda expensive, O(N^2)?

Reply:
Just for the confirmation-count=1 case, you mean?

Reply:
Yeah